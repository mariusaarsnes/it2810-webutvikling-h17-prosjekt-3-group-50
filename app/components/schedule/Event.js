import React from "react";
import {StyleSheet, TextInput, View, PixelRatio} from 'react-native';

export default class Event extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            eventTitle: "",
            eventDescription: ""
        };
        /*
        this.onTitleChange = this.onTitleChange.bind(this);
        this.onDescriptionChange = this.onDescriptionChange.bind(this);
        */
    }

    render() {
        return (
            <View style={styles.event}>
                <TextInput style={styles.eventTitle} placeholder={'Event title'}
                />
                <TextInput style={styles.eventDescription} placeholder={'Event description'}
                /> 
            </View>
        )
    }

    /*
    onTitleChange({target}) {
        this.setState({eventTitle: target.value, eventDescription: this.state.eventDescription});
    }

    onDescriptionChange({target}) {
        this.setState({eventTitle: this.state.eventTitle, eventDescription: target.value});
    }

    componentDidUpdate() {
        this.save();
    }

    componentDidMount() {
        const data = this.load();
        if (data !== null) {
            this.setState(() => data);
        } else {
            this.save();
        }
    }


    save() {
        localStorage.setItem(this.getSaveName(), JSON.stringify(this.state))
    }

    load() {
        return this.getSaveName() in localStorage
            ? JSON.parse(localStorage.getItem(this.getSaveName()))
            : null;
    }

    getSaveName() {
        return "" + this.props.day + this.props.slotId + this.props.id;
    }
    */
}

const styles = StyleSheet.create({
    event: {
        display: 'flex',
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        backgroundColor: 'white',
    },
    eventTitle: {
        flex: 1,
        borderBottomWidth: 1 / PixelRatio.get(),
    },
    eventDescription: {
        flex: 1
    }
})