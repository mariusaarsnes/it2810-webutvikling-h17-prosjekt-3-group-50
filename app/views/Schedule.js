import React from 'react';
import {View} from 'react-native';
import {DayGrid} from "../components/Schedule/DayGrid";

export class Schedule extends React.Component {

    render() {
        return(
            <View>
                <View>
                    <br/>
                    <br/><br/>
                    <DayGrid/>
                </View>
            </View>

        )
    }
}
