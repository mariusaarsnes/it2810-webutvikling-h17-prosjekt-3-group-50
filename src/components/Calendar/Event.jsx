import React from "react";

export class Event extends React.Component {
    render() {
        return (
            <div className={"event"}>
                <input className={"event-title"}/>
                <input className={"event-description"}/>
            </div>
        )
    }
}