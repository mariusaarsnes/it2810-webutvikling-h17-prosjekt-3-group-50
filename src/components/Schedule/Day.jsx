import React from 'react';
import {SlotGrid} from "./SlotGrid.jsx";

export class Day extends React.Component {

    constructor(props) {
        super(props);

    }
    render() {
        return(
            <div className={"day"}>
                <p className={"day-text"}>{this.props.day}</p>
                <SlotGrid day={this.props.day}/>
            </div>
        )
    }
}