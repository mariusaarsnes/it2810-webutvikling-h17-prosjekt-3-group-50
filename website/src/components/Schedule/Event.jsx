import React from "react";

export class Event extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            eventTitle: "",
            eventDescription: ""
        };

        this.onTitleChange = this.onTitleChange.bind(this);
        this.onDescriptionChange = this.onDescriptionChange.bind(this);
    }

    render() {
        return (
            <div className={"event"}>
                <input className={"event-title"}
                       value={this.state.eventTitle}
                       onChange={this.onTitleChange}
                />
                <input className={"event-description"}
                       value={this.state.eventDescription}
                       onChange={this.onDescriptionChange}
                />
            </div>
        )
    }

    onTitleChange({target}) {
        this.setState({eventTitle: target.value, eventDescription: this.state.eventDescription});
    }

    onDescriptionChange({target}) {
        this.setState({eventTitle: this.state.eventTitle, eventDescription: target.value});
    }

    componentDidUpdate() {
        this.save();
    }

    componentDidMount() {
        const data = this.load();
        if (data !== null) {
            this.setState(() => data);
        } else {
            this.save();
        }
    }


    save() {
        localStorage.setItem(this.getSaveName(), JSON.stringify(this.state))
    }

    load() {
        return this.getSaveName() in localStorage
            ? JSON.parse(localStorage.getItem(this.getSaveName()))
            : null;
    }

    getSaveName() {
        return "" + this.props.day + this.props.slotId + this.props.id;
    }
}