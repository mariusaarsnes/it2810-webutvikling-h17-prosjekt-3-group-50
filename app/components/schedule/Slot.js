import React from "react";
import {StyleSheet, Text, TextInput, View, PixelRatio} from 'react-native';
import Event from "./Event.js";

export default class Slot extends React.Component {
    constructor(props) {
        super(props);

        this.i = 0;

        this.state = {
            color: this.props.color,
            interval: this.props.interval
        };

        /*
        this.onIntervalChange = this.onIntervalChange.bind(this);
        this.generateNewEvent = this.generateNewEvent.bind(this);
        */
    }

    render() {
        return (
            <View style={[{backgroundColor: '' + this.state.color}, styles.slot]}>
                <View style={[{backgroundColor: '' + this.state.color}, styles.time]}>
                    <TextInput style={[styles.slotInterval, styles.slotIntervalStart]}
                               value={this.state.interval[0]}/>
                    <Text>-</Text>
                    <TextInput style={[styles.slotInterval, styles.slotIntervalEnd]}
                               value={this.state.interval[1]}/>
                </View>
                <View style={styles.text}>
                    <Event/>
                </View>
            </View>
        )
    }

    /*
        onIntervalChange({target}) {

            let classNames = target.className.split(" ");
            let tempInterval = [];
            if (classNames[1] === "slot-interval-start") {
                tempInterval.push(target.value);
                tempInterval.push(this.state.interval[1]);
            } else {
                tempInterval.push(this.state.interval[0]);
                tempInterval.push(target.value);
            }
            this.setState({color: this.state.color, interval: tempInterval});

        }


        componentDidUpdate() {
            this.save()
        }
        */

    /*
    When the component mounts we check to see if there is data saved in localStorage related to it.
    If not, then this is the first time the component is mounted and we save the state for later
    */
    /*
        componentDidMount() {
            const data = this.load();
            if (data !== null) {
                this.setState(() => data);
            } else {
                this.save();
            }
        }

        save() {
            localStorage.setItem(this.getSaveName(), JSON.stringify(this.state))
        }

        load() {
            return this.getSaveName() in localStorage
                ? JSON.parse(localStorage.getItem(this.getSaveName()))
                : null;
        }

        getSaveName() {
            return "" + this.props.day + this.props.id;
        }
    */
}

const styles = StyleSheet.create({
    slot: {
        flex: 1,
        display: 'flex',
        flexDirection: 'row',
        borderBottomColor: 'black',
        borderBottomWidth: 1 / PixelRatio.get()
    },
    text: {
        flex: 2

    },
    time: {
        flex: 1
    },

    slotInterval: {},
    slotIntervalStart: {},
    slotIntervalEnd: {}
});