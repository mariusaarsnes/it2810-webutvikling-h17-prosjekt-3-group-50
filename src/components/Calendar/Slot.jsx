import React from "react";
import {Event} from "./Event.jsx";

export class Slot extends React.Component {
    constructor(props) {
        super(props);

    }


    render() {
        return (
            <div className={"slot"} style={{backgroundColor: '' + this.props.color}} >
                <div>
                    <input className={"slot-interval slot-interval-start"} defaultValue={this.props.interval[0]}/>
                    -
                    <input className={"slot-interval slot-interval-end"} defaultValue={this.props.interval[1]} />
                </div>
                {
                    <Event/>
                }
            </div>
        )
    }
}