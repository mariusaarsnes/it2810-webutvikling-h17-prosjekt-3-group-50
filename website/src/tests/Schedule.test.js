//localStorage is not imported to be used as a class, but to be initiated with jest.
import localStorage from './localStorageMock';

import React from 'react';
import renderer from 'react-test-renderer';
import {Event} from "../components/Schedule/Event";
import {Slot} from "../components/Schedule/Slot";

/*
describe('Schedule Overview', () => {
    it('renders the whole Schedule application', () => {
        const instance = renderer.create(<Schedule />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
*/

describe('Schedule Components', () => {
    it('renders the Event Component', () => {
        const instance = renderer.create(
            <Event />
        );
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the Slot Component', () => {
        const instance = renderer.create(
            <Slot day={"Mo"}
                   id={1}
                   interval={0}
                   color={"#c7b9e5"}/>
        );
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    });

    /* Todo: Test other components that auto-generate colors
    *  - find out how to do this
    */
});