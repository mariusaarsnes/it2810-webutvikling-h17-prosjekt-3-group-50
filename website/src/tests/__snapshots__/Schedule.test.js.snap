// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule Components renders the Event Component 1`] = `
<div
  className="event"
>
  <input
    className="event-title"
    onChange={[Function]}
    value=""
  />
  <input
    className="event-description"
    onChange={[Function]}
    value=""
  />
</div>
`;

exports[`Schedule Components renders the Slot Component 1`] = `
<div
  className="slot"
  style={
    Object {
      "backgroundColor": "#c7b9e5",
    }
  }
>
  <div>
    <input
      className="slot-interval slot-interval-start"
      onChange={[Function]}
      value={undefined}
    />
    -
    <input
      className="slot-interval slot-interval-end"
      onChange={[Function]}
      value={undefined}
    />
  </div>
  <div
    className="event"
  >
    <input
      className="event-title"
      onChange={[Function]}
      value=""
    />
    <input
      className="event-description"
      onChange={[Function]}
      value=""
    />
  </div>
</div>
`;
